cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -345,25 +348,35 @@ protected void configure() {
         }
       }),
       new ProvisionerModule(),
-      new AbstractModule() {
+      new PrivateModule() {
         @Override
         protected void configure() {
+          // ArtifactRepositoryReader is required by DefaultArtifactRepository.
+          // Keep ArtifactRepositoryReader private to minimize the scope of the binding visibility.
+          bind(ArtifactRepositoryReader.class).to(LocalArtifactRepositoryReader.class).in(Scopes.SINGLETON);
           bind(ArtifactRepository.class)
             .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO))
             .to(DefaultArtifactRepository.class)
             .in(Scopes.SINGLETON);
+          expose(ArtifactRepository.class)
```

Comment:
Same here, you are exposing this binding anyway, why make it private

Reply:
This way we limit the visibility of RepoReader's binding, meaning others can bind RepoReader differently.